opm/grid/cpgrid/CpGrid.cpp

```diff
@@ -930,9 +930,25 @@ const CpGridFamily::Traits::LeafIndexSet& CpGrid::leafIndexSet() const
     return *current_view_data_->index_set_;
 }
 
-void CpGrid::globalRefine (int)
-{
-    std::cout << "Warning: Global refinement not implemented, yet." << std::endl;
+void CpGrid::globalRefine (int refCount)
+{
+    assert(refCount>-1);
+    // Throw if the grid has been already partially refined, i.e., there exist coarse cells with more than 6 faces.
+    // This is the case when a coarse cell has not been marked for refinement, but at least one of its neighboring cells
+    // got refined. Therefore, the coarse face that they share got replaced by refined-faces.
+    // if ((data_.size()>1) && )
```

Comment:
Why this comment? There is no throw here.

Reply:
I forgot to delete it. Just "an old thought"

Reply:
I forgot, it's fixed now, and tested!